internal/htmlForm/htmlForm.go

```diff
@@ -0,0 +1,106 @@
+package htmlForm
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/PuerkitoBio/goquery"
+	"golang.org/x/net/html"
+)
+
+// HtmlForm represents needed elements of an HTML Form.
+type HtmlForm struct {
+	// Action parsed from action form's field
+	Action string
+	// URL is the complete url where form will be posted
+	URL string
+	// Values contains htmlForm values to be submitted
+	Values url.Values
+}
+
+// ParseForms parses and returns all htmlForm elements beneath node.  Form values
+// include all input, select with the first possible value and submit buttons. The values of radio
+// and checkbox inputs are included only if they are checked.
+func ParseForms(node *html.Node) (forms []*HtmlForm) {
+	if node == nil {
+		return nil
+	}
+
+	doc := goquery.NewDocumentFromNode(node)
```

Comment:
`goquery.NewDocumentFromReader` can be also used. And no need to transform `io.Reader` to `*html.Node` by your own.